### ISSUE TITLE
 Docker: push error: xorm.log permission change to root:root by daily log rotation

### ISSUE THREAD

**Issue by gnboorse** [gogs/gogs#4990]
- Gogs version (or commit ref): 0.11.34.1122
- Git version: 2.11.3
- Operating system: Alpine (Docker container latest)
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist (usually found in `log/gogs.log`): from log/hooks/pre-receive.log

`2018/01/12 10:29:55 [FATAL] [...its/gogs/cmd/serv.go:88 setup()] SetEngine: Fail to create 'xorm.log': initFile: OpenFile '/app/gogs/log/xorm.log': open /app/gogs/log/xorm.log: permission denied`

## Description
This issue is hard to reproduce because it happens intermittently for me. Basically the issue is that sometimes when I attempt to perform a push to a git repository, I receive this error: 

```Counting objects: 2, done.
Compressing objects: 100% (1/1), done.
Writing objects: 100% (2/2), 238 bytes | 23.00 KiB/s, done.
Total 2 (delta 0), reused 0 (delta 0)
remote: Gogs: Internal error
To [url redacted]
! [remote rejected] master-> master (pre-receive hook declined)
```

We do not have any custom pre-receive hooks defined, and I have checked and double checked my app.ini configuration.

In the pre-receive.log file it gives me this:

`2018/01/12 10:29:55 [FATAL] [...its/gogs/cmd/serv.go:88 setup()] SetEngine: Fail to create 'xorm.log': initFile: OpenFile '/app/gogs/log/xorm.log': open /app/gogs/log/xorm.log: permission denied`

Which I can confirm with an `ls -ltrh /app/gogs/log`
```
-rw-r-----    1 git      git       108.6K Jan 11 00:00 gogs.log.2018-01-10
-rw-r-----    1 git      git          444 Jan 11 14:47 xorm.log.2018-01-11
-rw-r-----    1 git      git        66.9K Jan 12 00:07 gogs.log.2018-01-11
-rw-r-----    1 root     root           0 Jan 12 05:01 xorm.log
drwxr-xr-x    2 git      git         4.0K Jan 12 10:29 hooks
-rw-r-----    1 git      git        27.0K Jan 12 14:47 gogs.log
```
If I run a `chown -R git /app/gogs/log` of course it resolves the issue, but then again at midnight the permissions change.

Has anyone else run into this issue?

For now my workaround is to add the chown command to a cron job, but I don't like that.

...

**Comment by rmxhaha** [comment]
I ran into this problem just once just now.
I also don't know why.

**Comment by gnboorse** [comment]
@rmxhaha Hey I actually have an update on this: I think it might be related to the backups that I run.

Every night I have a backup script that runs `/app/gogs/gogs backup` and copies the zip file elsewhere. I've only ever noticed the file permissions issue to occur after that so I'm guessing the backup utility has some problem.

As I mentioned earlier, I just run `chown -R git /app/gogs/log` every night to fix this.

**Comment by SteveCliff** [comment]
I guess it doesn't really help as I haven't managed to track down why at all, but yes, I get the same problem as above ... and yes I "solve" by running a cron job each morning to reset the permission before the devs start work :)

**Comment by rmxhaha** [comment]
If I remember correctly, I just moved a gogs data into another VPS and that happened. So I am positive it must be because of the backup procedure. 

I don't do regular restore like you guys so I got this problem only once and maybe later if I move to another vps.

**Comment by CodeSmolder** [comment]
This seams to be the same issue: #5294

**Comment by vision-ken** [comment]
How is it going？

**Comment by vision-ken** [comment]
@gnboorse 
this is right:
```shell
$ chown -R git /app/gogs/log/
```
but not:
```shell
$ chown -R git /app/gogs/log
```

**Comment by vision-ken** [comment]
I made a [docker image](https://hub.docker.com/r/prong/gogs/) with features:
- Auto backup at 00:30 everday.
- Reserve the latest 5 backup files.
- Resolved issues related to backup and recovery.

**Comment by lgsdp** [comment]
New to Github so apologies if I'm replying in the wrong place.

Running Gogs 0.12.3 via Windows service and when the scripted backup runs we encounter the same error. I cannot see any updates for this bug since 2018, is it still on the list to be addressed?

2020/11/23 14:25:23 [FATAL] [gogs.io/gogs/gogs.go:40 main()] Failed to start application: set engine: create 'xorm.log': init: init rotation: rename rotate file: rename D:\Misc\gogs\log\xorm.log D:\Misc\gogs\log\xorm.log.2020-11-05: The process cannot access the file because it is being used by another process.

I've tried running the backup script at various times of the day but it always fails unless I restart the Gogs service just before hand. We've been using Gogs for over two years and the issue only started when we updated to 0.12.2.

**Comment by unknwon** [comment]
@lgsdp Hi, I believe this original issue ("unable to push") was also reported in https://github.com/gogs/gogs/issues/6316#issuecomment-699775038 and patched with 0.12.3.

(This issue seems having the same root cause as #6316, so closing)

However, the "scripted backup" you mentioned seems another place we need to take case of, I filed https://github.com/gogs/gogs/issues/6428 for this.
